Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift

```diff
@@ -36,7 +36,13 @@ protocol AssignmentDetailsViewProtocol: SubmissionButtonViewProtocol {
     func showSubmitAssignmentButton(title: String?)
 }
 
-class AssignmentDetailsPresenter {
+class AssignmentDetailsPresenter: PageViewLoggerPresenterProtocol {
+
+    var pageViewEventName: String {
+        guard let a = assignments.first else { return "\(String(describing: self))"}
```

Comment:
seems like it would happen often that the viewDidAppear will happen before the assignment is loaded. We have thr courseID & assignmentID as well as the baseURL from the session. Perhaps it would be better if we always constructed the url, instead of relying on the core data object.

Reply:
great suggestion